ublox/src/ubx_packets/packets.rs

```diff
@@ -1939,111 +1941,123 @@ struct RxmRtcm {
 #[ubx(class = 0x10, id = 0x02, max_payload_len = 1240)]
 struct EsfMeas {
     time_tag: u32,
+    flags: u16,
+    id: u16,
+    #[ubx(
+        map_type = EsfMeasDataIter,
+        from = EsfMeasDataIter::new,
+        size_fn = data_len,
+        is_valid = EsfMeasDataIter::is_valid,
+        may_fail,
+    )]
+    data: [u8; 0],
     #[ubx(
-        map_type = EsfMeasInfo,
-        from = EsfMeasInfo::new,
-        flatten,
-        flat_fields = ["flags", "u16", "id", "u16", "data", "U32Iter"]
+        map_type = Option<u32>,
+        from = EsfMeas::calib_tag,
+        size_fn = calib_tag_len,
     )]
-    //, "calib_tag", "Option<u32>"
-    info: [u8; 0],
+    calib_tag: [u8; 0],
 }
 
-#[derive(serde::Serialize, Debug)]
-pub struct EsfMeasInfo<'a> {
-    pub flags: u16,
-    pub id: u16,
-    pub data: U32Iter<'a>,
-    pub calib_tag: Option<u32>,
+impl EsfMeas {
+    fn calib_tag(bytes: &[u8]) -> Option<u32> {
+        bytes.try_into().ok().map(u32::from_le_bytes)
+    }
 }
 
-#[ubx_iter]
-pub struct U32Iter<'a>(core::slice::ChunksExact<'a, u8>);
+impl<'a> EsfMeasRef<'a> {
+    fn data_len(&self) -> usize {
+        ((self.flags() >> 11 & 0x1f) as usize) * 4
+    }
+
+    fn calib_tag_len(&self) -> usize {
+        if self.flags() & 0x8 != 0 {
+            4
+        } else {
+            0
+        }
+    }
+}
```

Comment:
unlike the other functions you add via attributes these ones are actually attached to the PacketRef